tests/feature_resolution_editions.js

```diff
@@ -0,0 +1,300 @@
+/*
+edition = "2023";
+
+option features.amazing_feature = A;
+option (mo_single_msg).nested.value = "x";
+service MyService {
+    option features.amazing_feature = E;
+    message MyRequest {};
+    message MyResponse {};
+    rpc MyMethod (MyRequest) returns (MyResponse) {
+        option features.amazing_feature = L;
+    };
+}
+
+message Message {
+    option features.amazing_feature = B;
+
+    string string_val = 1;
+    string string_repeated = 2 [features.amazing_feature = F];
+
+    uint64 uint64_val = 3;
+    uint64 uint64_repeated = 4;
+
+    bytes bytes_val = 5;
+    bytes bytes_repeated = 6;
+
+    SomeEnum enum_val = 7;
+    SomeEnum enum_repeated = 8;
+
+    extensions 10 to 100;
+    extend Message {
+        int32 bar = 10 [features.amazing_feature = I];
+    }
+
+    message Nested {
+        option features.amazing_feature = H;
+        int64 count = 9;
+    }
+
+    enum SomeEnumInMessage {
+        option features.amazing_feature = G;
+        ONE = 11;
+        TWO = 12;
+    }
+    
+    oneof SomeOneOf {
+        option features.amazing_feature = J;
+        int32 a = 13;
+        string b = 14;
+    }
+
+    map<string,int64> int64_map = 15;
+}
+
+extend Message {
+    int32 bar = 16 [features.amazing_feature = D];
+}
+
+enum SomeEnum {
+    option features.amazing_feature = C;
+    ONE = 1 [features.amazing_feature = K];
+    TWO = 2;
+}
+*/
+var tape = require("tape");
+
+var protobuf = require("..");
+
+var protoEditions2023 = `edition = "2023";`;
+
+var proto2 = `syntax = "proto2";`;
+
+var proto3 = `syntax = "proto3";`;
+
+var editions2023Defaults = {enum_type: 'OPEN', field_presence: 'EXPLICIT', json_format: 'ALLOW', message_encoding: 'LENGTH_PREFIXED', repeated_field_encoding: 'PACKED', utf8_validation: 'VERIFY'}
+var proto2Defaults = {enum_type: 'CLOSED', field_presence: 'EXPLICIT', json_format: 'LEGACY_BEST_EFFORT', message_encoding: 'LENGTH_PREFIXED', repeated_field_encoding: 'EXPANDED', utf8_validation: 'NONE'}
+var proto3Defaults = {enum_type: 'OPEN', field_presence: 'IMPLICIT', json_format: 'ALLOW', message_encoding: 'LENGTH_PREFIXED', repeated_field_encoding: 'PACKED', utf8_validation: 'VERIFY'}
+
+tape.test("feature resolution defaults", function(test) {
+    var rootEditions = protobuf.parse(protoEditions2023).root;
+    rootEditions.resolveAll();
```

Comment:
Do we need to call resolveAll in all these tests?  This worries me a bit, because it means any place where we *haven't* called resolveAll yet will see the wrong features

Reply:
Isn't that the behavior we want? Since features shouldn't be resolved while parsing? Not we're only calling `protobuf.parse` beforehand.

Reply:
They shouldn't be resolved *while* parsing, but if this is a public API it seems like a footgun for users

Reply:
Ok, what exactly do you think the behavior should be? Since we're resolving features during `resolve`, we do need to call it right now. What do you recommend doing instead?

Reply:
Let's take this offline, this is just a design question